cpp/oneapi/dal/algo/knn/backend/cpu/infer_kernel_brute_force_classification.cpp

```diff
@@ -14,21 +14,92 @@
 * limitations under the License.
 *******************************************************************************/
 
+#include <daal/src/algorithms/k_nearest_neighbors/bf_knn_classification_predict_kernel.h>
+
 #include "oneapi/dal/algo/knn/backend/cpu/infer_kernel.hpp"
+#include "oneapi/dal/algo/knn/backend/model_impl.hpp"
+#include "oneapi/dal/algo/knn/backend/distance_impl.hpp"
+#include "oneapi/dal/backend/interop/common.hpp"
 #include "oneapi/dal/backend/interop/error_converter.hpp"
-#include "oneapi/dal/detail/common.hpp"
+#include "oneapi/dal/backend/interop/table_conversion.hpp"
+
+#include "oneapi/dal/table/row_accessor.hpp"
 
 namespace oneapi::dal::knn::backend {
 
 using dal::backend::context_cpu;
-template <typename Float, typename Task>
-struct infer_kernel_cpu<Float, method::brute_force, Task> {
-    infer_result<Task> operator()(const context_cpu &ctx,
-                                  const detail::descriptor_base<Task> &desc,
-                                  const infer_input<Task> &input) const {
-        throw unimplemented(
-            dal::detail::error_messages::knn_brute_force_method_is_not_implemented_for_cpu());
-        return infer_result<Task>();
+using descriptor_t = detail::descriptor_base<task::classification>;
+
+namespace daal_knn = daal::algorithms::bf_knn_classification;
+namespace interop = dal::backend::interop;
+
+template <typename Float, daal::CpuType Cpu>
+using daal_knn_bf_kernel_t =
+    daal_knn::prediction::internal::KNNClassificationPredictKernel<Float, Cpu>;
+
+template <typename Float>
+static infer_result<task::classification> call_daal_kernel(const context_cpu &ctx,
+                                                           const descriptor_t &desc,
+                                                           const table &data,
+                                                           model<task::classification> m) {
+    const std::int64_t row_count = data.get_row_count();
+
+    auto arr_labels = array<Float>::empty(1 * row_count);
+
+    const auto daal_data = interop::convert_to_daal_table<Float>(data);
+    const auto daal_labels = interop::convert_to_daal_homogen_table(arr_labels, row_count, 1);
+
+    const auto data_use_in_model = daal_knn::doNotUse;
+    daal_knn::Parameter original_daal_parameter(
+        dal::detail::integral_cast<std::size_t>(desc.get_class_count()),
+        dal::detail::integral_cast<std::size_t>(desc.get_neighbor_count()),
+        data_use_in_model);
+
+    daal_knn::prediction::internal::KernelParameter daal_parameter;
+    daal_parameter.nClasses = original_daal_parameter.nClasses;
+    daal_parameter.k = original_daal_parameter.k;
+    daal_parameter.dataUseInModel = original_daal_parameter.dataUseInModel;
```

Comment:
dataUseInModel , engine no need for knn-brute. remove these parameters.

Reply:
done